projects/fsastorefrontlib/src/cms-components/b2b/unit/assignments/cells/remove-product-cell/remove-product-cell.component.ts

```diff
@@ -0,0 +1,57 @@
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
+import {
+  CellComponent,
+  CurrentUnitService,
+} from '@spartacus/organization/administration/components';
+import {
+  OutletContextData,
+  TableDataOutletContext,
+} from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ProductAssignmentService } from '../../../../../../core/product-assignment/facade/product-assignment.service';
+
+@Component({
+  selector: 'cx-fs-remove-product-cell',
+  templateUrl: './remove-product-cell.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RemoveProductCellComponent extends CellComponent
+  implements OnDestroy {
+  constructor(
+    protected outlet: OutletContextData<TableDataOutletContext>,
+    protected productAssignmentService: ProductAssignmentService,
+    protected currentUnitService: CurrentUnitService
+  ) {
+    super(outlet);
+  }
+
+  private subscription = new Subscription();
+
+  currentUnit$ = this.currentUnitService.item$;
+
+  removeProduct() {
+    this.subscription.add(
+      this.currentUnit$
```

Comment:
Tried to pass the data from this observable in the template since it's already available there, but the view doesn't reflect the changes for some reason. On refresh, the state is correctly displayed, but that is unacceptable, so follow-up task will be created to investigate this.

Reply:
Fixed, works now